libs/utils/trace_analysis.py

```diff
@@ -987,8 +1009,9 @@ def getClusterFrequencyResidency(self, cluster):
         #     freq_active[t] == 1 if at time t the frequency is f
         #     freq_active[t] == 0 otherwise
         available_freqs = sorted(cluster_freqs.unique())
-        new_idx = sorted(cluster_freqs.index.tolist() + \
-                         cluster_active.index.tolist())
+        # Use set() to remove possible duplicate indexes
+        new_idx = sorted(set(cluster_freqs.index.tolist() + \
```

Comment:
You are right. I removed the `set()` instruction and provided a check in the crop function to make sure that no duplicate indexes are generated (it could happen if the user specifies an x range with a value that is exactly the same as a time stamp in the DataFrame).